commands/localBuild.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	newLocalBuildCmd := newLocalBuildCmd()
+	// Setup flags that are used by multiple commands (variables defined in faas.go)
+	newLocalBuildCmd.Flags().StringVar(&image, "image", "", "Docker image name to build")
+	newLocalBuildCmd.Flags().StringVar(&handler, "handler", "", "Directory with handler for function, e.g. handler.js")
+	newLocalBuildCmd.Flags().StringVar(&functionName, "name", "", "Name of the deployed function")
+	newLocalBuildCmd.Flags().StringVar(&language, "lang", "", "Programming language template")
+
+	// Setup flags that are used only by this command (variables defined above)
+	newLocalBuildCmd.Flags().BoolVar(&nocache, "no-cache", false, "Do not use Docker's build cache")
+	newLocalBuildCmd.Flags().BoolVar(&squash, "squash", false, `Use Docker's squash flag for smaller images [experimental] `)
+	newLocalBuildCmd.Flags().IntVar(&parallel, "parallel", 1, "Build in parallel to depth specified.")
+	newLocalBuildCmd.Flags().BoolVar(&shrinkwrap, "shrinkwrap", false, "Just write files to ./build/ folder for shrink-wrapping")
+	newLocalBuildCmd.Flags().StringArrayVarP(&buildArgs, "build-arg", "b", []string{}, "Add a build-arg for Docker (KEY=VALUE)")
+	newLocalBuildCmd.Flags().StringArrayVarP(&buildOptions, "build-option", "o", []string{}, "Set a build option, e.g. dev")
+	newLocalBuildCmd.Flags().Var(&tagFormat, "tag", "Override latest tag on function Docker image, accepts 'latest', 'sha', 'branch', or 'describe'")
+	newLocalBuildCmd.Flags().StringArrayVar(&buildLabels, "build-label", []string{}, "Add a label for Docker image (LABEL=VALUE)")
+	newLocalBuildCmd.Flags().StringArrayVar(&copyExtra, "copy-extra", []string{}, "Extra paths that will be copied into the function build context")
+	newLocalBuildCmd.Flags().BoolVar(&envsubst, "envsubst", true, "Substitute environment variables in stack.yml file")
+	newLocalBuildCmd.Flags().BoolVar(&quietBuild, "quiet", false, "Perform a quiet build, without showing output from Docker")
+	newLocalBuildCmd.Flags().BoolVar(&disableStackPull, "disable-stack-pull", false, "Disables the template configuration in the stack.yml")
```

Comment:
Can create functions for the flags and options for avoiding repetition in the `build.go`, `local_run.go` and `localBuild.go`, right?